core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/config/BaseSailConfig.java

```diff
@@ -71,6 +83,9 @@ public void parse(Model graph, Resource implNode) throws SailConfigException {
 		super.parse(graph, implNode);
 
 		try {
+			Models.objectLiteral(graph.getStatements(implNode, BaseSailSchema.DEFAULT_QUERY_EVALUATION_MODE, null))
+					.ifPresent(qem -> setDefaultQueryEvaluationMode(
+							QueryEvaluationMode.valueOf(qem.stringValue().toUpperCase(Locale.ROOT))));
```

Comment:
I've deliberately designed this to be forgiving, and optional. Why would you want it to throw an exception?

Reply:
I think we should throw an exception if the value is invalid. That way users will get immediate feedback if they for instance have a typo.

The exception we throw can include a list of all the allowed values, we can probably generate the list at runtime by using Arrays.toString(...). 

I still think it's easier to be strict on case sensitivity. We can always be more lenient in the future, but if we start by being case insensitive we can't go back.

Reply:
Checked the Java docs and I think that the valueOf(...) method should throw an IllegalArgumentException if the string doesn't have a match. I think it would be helpful for our users if we caught that exception and threw a more helpful exception. Maybe just add IllegalArgumentException to the catch clause where we are throwing the config exception? 

If there is no value present then I agree that we shouldn't throw any exception. 

I would still like to be strict on case sensitivity.

Reply:
I'm not going back on the case leniency as I'd have to weed through all templates again to fix the currently set values (which are all lower-case). But good suggestion on catching the IllegalArgumentException.

Reply:
> I'm not going back on the case leniency as I'd have to weed through all templates again to fix the currently set values (which are all lower-case). But good suggestion on catching the IllegalArgumentException.

I've slept on it, I'll also remove the case leniency (and while I'm at it I'll introduce some additional testing).